### ISSUE TITLE
Builders are not properly "saved" when nested types with the same last component are used

### ISSUE THREAD

**Issue by zeusent** [johnpatrickmorgan/NavigationBackport#3]
For example take these two structures:

```
struct Section {

    struct Item {
        ...
    }

    var items: [Section.Item]

}

struct SomeOtherTypeWithNestedItemStruct {

    struct Item {
        ...
    }

    var items: [SomeOtherTypeWithNestedItemStruct.Item]

}
```

If you later use the two `Item` structs as types for `nbNavigationDestination` they will override each other's builders and linking will fail:

```
NBNavigationStack(path: $somePathBinding) {
    SomeView()
        .nbNavigationDestination(for: Section.Item.self, destination: { item in
            SomeItemView(item: item)
        })
        .nbNavigationDestination(for: SomeOtherTypeWithNestedItemStruct.Item.self) { item in
            AnotherItemView(item: item)
        }
}
```

**Comment by zeusent** [comment]
Maybe try `builders[String(reflecting: T.self)]` instead of `builders["\(T.self)"]` to store those builders...

**Comment by johnpatrickmorgan** [comment]
Thanks for finding this issue @zeusent, and for the suggested change. It looks like the best approach, I'll make the necessary changes...

**Comment by johnpatrickmorgan** [comment]
Released in [v0.1.2](https://github.com/johnpatrickmorgan/NavigationBackport/releases/tag/0.1.2)
